Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs

```diff
@@ -698,3 +700,40 @@ clearOnRideStuckDrivers merchantShortId = do
       )
       driverInfos
   return Common.ClearOnRideStuckDriversRes {driverIds = driverIds}
+
+---------------------------------------------------------------------
+
+driverAadhaarInfoByPhone :: ShortId DM.Merchant -> Text -> Flow Common.DriverAadhaarInfoRes
+driverAadhaarInfoByPhone merchantShortId phoneNumber = do
+  merchant <- findMerchantByShortId merchantShortId
+  mobileNumberHash <- getDbHash phoneNumber
+  driver <- QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash merchant.id >>= fromMaybeM (InvalidRequest "Person not found")
+  res <- AV.findByDriverId driver.id
+  case res of
+    Just aadhaarData -> do
+      pure
+        Common.DriverAadhaarInfoRes
+          { driverName = aadhaarData.driverName,
+            driverGender = aadhaarData.driverGender,
+            driverDob = aadhaarData.driverDob,
+            driverImage = aadhaarData.driverImage
+          }
+    Nothing -> throwError $ InvalidRequest "no aadhaar data is found"
+
+---------------------------------------------------------------------
+
+---------------------------------------------------------------------
+
+updateDriverAadhaar :: ShortId DM.Merchant -> Text -> Common.UpdateDriverDataReq -> Flow APISuccess
+updateDriverAadhaar merchantShortId phoneNumber req = do
+  mobileNumberHash <- getDbHash phoneNumber
+  aadhaarNumberHash <- getDbHash req.driverAadhaarNumber
+  aadhaarInfo <- AV.findByAadhaarNumberHash aadhaarNumberHash
```

Comment:
Aadhaar info will always be there, right?

Reply:
As we are using the same table.

Reply:
yes but let say that aadhaar is already linked with some other account then we  should not allow to update that